crates/bevy_render/src/color/mod.rs

```diff
@@ -183,6 +183,14 @@ impl Color {
         Err(HexColorError::Length)
     }
 
+    /// New `Color` from sRGB colorspace.
+    ///
+    /// eg. `Color::rgba_hex_u32(0xFF0000FF)`
+    pub fn rgba_hex_u32(input: u32) -> Color {
+        let [r, g, b, a] = input.to_le_bytes();
+        Self::rgba_u8(r, g, b, a)
```

Comment:
Would we be better off making the default behaviour actually being:
```rust
let [_surplus, r, g, b] = input.to_le_bytes();
debug_assert_ne!(_surplus, 0);
Self::rgb(r, g, b)
```
with another function `[name]_alpha` for this version?

That avoids the issue with hex codes mentioned in my other issue, but obviously ignores some of the u32 range.

Reply:
When we are using ... (0xFF0000FF) we should provide an alpha, because when creating colors with u32 we will always create a `rgba` and if no alpha channel is entered then the alpha will be 0, so we should always see colors that start or end with `ff`.

This is different from Color::hex, which allows rgb and rgba to be supplied.

```
Color::hex("fff"); // alpha = 1.0
```

---

`Flutter` example:

```dart
Color(0xFF0000);  // Color.fromARGB(0, 255, 0, 0);
```

Reply:
Yes?

But surely, realistically the vast majority of calls to this function will need an opaque colour, rather than a fully invisible one.

And the vast majority of sources of hex colour codes will exclude the alpha channel, assuming it will be treated as opaque. But this function currently requires every receiver of such a code to add the FF, or else get weirdly transposed colours. I'm saying that maybe we shouldn't do that, and just ignore one byte of the u32.

Reply:
> But surely, realistically the vast majority of calls to this function will need an opaque colour, rather than a fully invisible one.

Totally agree.

---

I still think we can't ignore any bytes, because u32 does contain four channels of rgba, and what if the user really wants a completely invisible color?

```rust
Color::rgba_hex_u32(0xFFFFFF);  // alpha = 1.0

// 🙅‍♂️
Color::rgba_hex_u32(0xFFFFFF00);  // alpha = 1.0
// But what we want is alpha = 0
```

Reply:
I'm not sure I follow

Just because u32 contains 4 bytes, doesn't mean those 4 bytes have to map to four channels. I'm aiming to get to the best UX here, which I don't think needing to specify the alpha channel for the reasons laid out above.

Changing the behaviour of the 'primary' method to give it better UX doesn't mean we can't have another function which has this current behaviour (or has the alpha channel first). Indeed, that was the original suggestion at which this discussion started.